### DISCUSSION TITLE
[Tables] Using Intl extension for numeric columns

### DISCUSSION THREAD

**Discussion by Sicklou** [filamentphp/filament#8271] [Ideas]
Hi everyone,

I was wondering if it can be a good idea to have the Intl NumberFormatter extension handling numeric values in the table package by default. (see [NumberFormatter docs](https://www.php.net/manual/en/class.numberformatter.php))

Most of time, we have tables that display numeric with consistency between tables. It can be cumbersome to choose everytime the formatting and not very DX friendly.

With the existing way, if I want to display numeric values for french users, I have to do on every ,numeric column : 
```php
Columns\TextColumn::make('quantity')
    ->numeric(
        decimalPlaces: 0,
        decimalSeparator: ',',
        thousandsSeparator: ' ',
    )
    ->label('Quantity'),
```

What if by default we can simply set the numeric part like that
```php
Columns\TextColumn::make('quantity')
    ->numeric()
    ->label('Quantity'),
```

Using intl, if some people built products with different users around the world, it can adapt to it.

This is what I tried on the fly : 
```php 
public function numeric(bool $locale = true, int | Closure | null $decimalPlaces = null, string | Closure | null $decimalSeparator = '.', string | Closure | null $thousandsSeparator = ','): static
{
    if ($locale) {
        $this->formatStateUsing(static function (TextColumn $column, $state): ?string {
            return (new \NumberFormatter(config('app.locale'), \NumberFormatter::DECIMAL))->format($state);
        });

        return $this;
    }

    $this->formatStateUsing(static function (TextColumn $column, $state) use ($decimalPlaces, $decimalSeparator, $thousandsSeparator): ?string {
        if (blank($state)) {
            return null;
        }

        if (! is_numeric($state)) {
            return $state;
        }

        if ($decimalPlaces === null) {
            return format_number($state);
        }

        return number_format(
            $state,
            $column->evaluate($decimalPlaces),
            $column->evaluate($decimalSeparator),
            $column->evaluate($thousandsSeparator),
        );
    });

    return $this;
}
```

What do you think about this ?

**Comment by danharrin** [comment]
To avoid breaking changes it could be `locallyNumeric()` or something

**Reply by danharrin** [reply]
Also the same for infolist entry classes
